File: js/containsDuplicate.js
// LeetCode - 217. Contains Duplicate (https://leetcode.com/problems/contains-duplicate/)

var containsDuplicate = function(nums) {
  var setNumbers = new Set()
  for(let i = 0; i<= nums.length; i++){
      if(setNumbers.size == 0){
          setNumbers.add(nums[i])
          continue
      }

      if (setNumbers.has(nums[i])){
          return true
      }

      setNumbers.add(nums[i])
  }

  return false
};

console.log(containsDuplicate(nums = [1,2,3,1]))

File: arrays/containsDuplicate.go
package arrays

import "fmt"

// LeetCode 217. Contains Duplicate

func containsDuplicate(nums []int) bool {
	setNumbers := make(map[int]struct{})

	for _, num := range nums {
			if _, exists := setNumbers[num]; exists {
					return true
			}
			setNumbers[num] = struct{}{}
	}

	return false
}

func main() {
	nums := []int{1, 2, 3, 4, 5, 1}
	fmt.Println(containsDuplicate(nums)) // Output: true
}